src/utils/index.js

```diff
@@ -206,6 +211,88 @@ const FormioUtils = {
   },
 
   /**
+   * Extension of standard #parseFloat(value) function, that also clears input string.
+   *
+   * @param {any} value
+   *   The value to parse.
+   *
+   * @returns {Number}
+   *   Parsed value.
+   */
+  parseFloat: function(value) {
+    return parseFloat(_isString(value)
+      ? value.replace(/[^\de.+-]/gi, '')
+      : value);
+  },
+
+  /**
+   * Formats provided value in way how Currency component uses it.
+   *
+   * @param {any} value
+   *   The value to format.
+   *
+   * @returns {String}
+   *   Value formatted for Currency component.
+   */
+  formatAsCurrency: function(value) {
+    const parsedValue = this.parseFloat(value);
+
+    if (_isNaN(parsedValue)) {
+      return '';
+    }
+
+    const parts = _round(parsedValue, 2)
```

Comment:
This bit of code is crazy. I would feel so much better if this had some Unit tests within the ```utils.spec.js``` file to accompany how this is supposed to behave.